### PR TITLE
Resolve: [Ledger] Add wallet index input for RUNE

### PR DESCRIPTION
**Pull request by gromxyz**
Closes #1888

### CODE REVIEW

File: src/main/api/ledger/transaction.ts [hunk 1]
```diff
@@ -24,7 +24,14 @@ export const sendTx = async ({
...
     switch (chain) {
       case THORChain:
-        res = await THOR.send({ transport, network, recipient, amount, memo })
+        res = await THOR.send({
+          transport,
+          network,
+          recipient,
+          amount,
+          memo,
+          walletIndex: walletIndex ? walletIndex : 0
```
Comment (veado) [inline src/main/api/ledger/transaction.ts]: [nitpick] You can already set `walletIndex = 0` params of `sendTx`

File: src/renderer/components/wallet/settings/WalletSettings.tsx [hunk 2]
```diff
@@ -72,7 +83,16 @@ export const WalletSettings: React.FC<Props> = (props): JSX.Element => {
   const [showRemoveWalletModal, setShowRemoveWalletModal] = useState(false)
   const [showQRModal, setShowQRModal] = useState<O.Option<{ asset: Asset; address: Address }>>(O.none)
   const closeQrModal = useCallback(() => setShowQRModal(O.none), [setShowQRModal])
-  const [walletIndex, setWalletIndex] = useState<number>(0)
+  const [walletIndex, setWalletIndex] = useState({
```
Comment (veado) [inline src/renderer/components/wallet/settings/WalletSettings.tsx]: [nitpick]
```ts
useState<Record<Chain, number>>(...)
```

File: src/renderer/components/wallet/settings/WalletSettings.tsx [hunk 3]
```diff
@@ -105,18 +125,20 @@ export const WalletSettings: React.FC<Props> = (props): JSX.Element => {
     (chain: Chain, { type: walletType, address: addressRD }: WalletAddressAsync) => {
       const renderAddLedger = (chain: Chain, loading: boolean) => (
```
Comment (veado) [inline src/renderer/components/wallet/settings/WalletSettings.tsx:126]: [nitpick] 
```ts
const walletIndex = walletIndex[chain]
```
_Just to be more DRY_

File: src/main/api/ledger/transaction.ts [hunk 4]
```diff
@@ -58,14 +65,15 @@ export const deposit = async ({
   chain,
   network,
   amount,
-  memo
+  memo,
+  walletIndex
```
Comment (veado) [inline src/main/api/ledger/transaction.ts:69]: `walletIndex = 0` (Similar to previous comment)

**Review by veado — CHANGES_REQUESTED** [review]
Looks good! _Just few nitpick..._

File: src/renderer/components/wallet/settings/WalletSettings.tsx (hunk 3 above)
Comment (gromxyz) [reply src/renderer/components/wallet/settings/WalletSettings.tsx:126]: But then it gets a little bit problematic for:
`setWalletIndex({ ...walletIndex, [chain]: +value })`

Comment (gromxyz) [reply src/renderer/components/wallet/settings/WalletSettings.tsx:126]: Since it would refer to the wrong one and then this logic would need to be changed

Comment (veado) [reply src/renderer/components/wallet/settings/WalletSettings.tsx:126]: Ok

**Review by veado — APPROVED** [review]
:+1:

**Pull request merged**